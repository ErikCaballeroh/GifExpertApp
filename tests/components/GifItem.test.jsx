import { render, screen } from "@testing-library/react"
import { GifItem } from "../../src/components/GifItem"

describe('Pruebas en <GifItem/>', () => {

    const image = {
        title: 'Titulo de imagen',
        url: 'https://one-punch.com/saitama.jgp',
        id: 'aorechu134',
    }

    test('debe de hacer match con el snapshoot', () => {
        const { container } = render(<GifItem {...image} />);
        expect(container).toMatchSnapshot();
    });

    test('debe de mostrar la imagen con el URL y el ALT indicado', () => {
        render(<GifItem {...image} />);
        const { title, url } = image;
        // screen.debug();
        // expect(screen.getByRole('img').src).toBe(image.url);
        // expect(screen.getByRole('img').alt).toBe(image.title);
        const { src, alt } = screen.getByRole('img');
        expect(src).toBe(url);
        expect(alt).toBe(title);
    });

    test('debe de mostrar el titulo en el componente', () => {
        render(<GifItem {...image} />);
        expect(screen.getByText(image.title)).toBeTruthy();
    })

})