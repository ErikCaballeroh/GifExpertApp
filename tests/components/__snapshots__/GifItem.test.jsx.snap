// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifItem/> debe de hacer match con el snapshoot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Titulo de imagen"
      src="https://one-punch.com/saitama.jgp"
    />
    <p>
      Titulo de imagen
    </p>
  </div>
</div>
`;
